src/features/klant/klant-details/fetch-klant.ts

```diff
@@ -11,65 +11,88 @@ import {
   type Systeem,
 } from "@/services/environment/fetch-systemen";
 import type { Klant } from "@/services/openklant/types";
-import { mapKlantToKlantIdentifier } from "@/features/contact/shared";
-import { findBedrijfInHandelsRegister, type Bedrijf } from "@/services/kvk";
+
+import { useContactmomentStore } from "@/stores/contactmoment";
+import { searchBedrijvenInHandelsRegisterByRsin } from "@/services/kvk";
 import { enforceOneOrZero } from "@/services";
 
-export const fetchKlant = async ({
-  id,
+import type { KlantIdentificator } from "@/features/contact/types";
+
+export const fetchKlantByInternalId = async ({
+  internalId,
   systemen,
   defaultSysteem,
 }: {
-  id: string;
+  internalId: string;
   systemen: Systeem[];
   defaultSysteem: Systeem;
 }): Promise<Klant | null> => {
-  const klant = await fetchKlantById(id, defaultSysteem);
-  if (!klant) return null;
-  if (heeftContactgegevens(klant)) return klant;
-  if (!systemen.length) return klant;
-
-  if (!klant.bsn) {
-    // For non-natural persons, we have EITHER an RSIN OR a Chamber of Commerce number (kvknummer),
-    // depending on whether the default system is ok1 or ok2.
-    // To translate this to the other systems,
-    // we need BOTH. So we first need to fetch the company again.
-    const bedrijf = await findBedrijfInHandelsRegister({
-      kvkNummer: klant.kvkNummer || klant.nietNatuurlijkPersoonIdentifier || "",
-      vestigingsnummer: klant.vestigingsnummer || "",
-      rsin: klant.rsin || klant.nietNatuurlijkPersoonIdentifier,
-    }).then(enforceOneOrZero);
-
-    if (!bedrijf) return klant;
-
-    klant.kvkNummer = bedrijf.kvkNummer;
-    klant.rsin = bedrijf.rsin;
-  }
+  const store = useContactmomentStore();
```

Comment:
Reactive dingen in een async context word ik altijd een beetje nerveus van :) Dat gaat vaak goed en soms fout en is dan een nachtmerrie om te debuggen. Misschien kan deze functie een parameter krijgen die de KlantDetails.vue component uit de store haalt?

Reply:
d1cdd1111bfcd919db0163d3ef841ca7b3ed2f6d